vividus-extension-selenium/src/main/java/org/vividus/ui/action/search/AbstractElementAction.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.ui.action.search;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.WebElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.ui.action.IExpectedConditions;
+import org.vividus.ui.action.IWaitActions;
+
+public abstract class AbstractElementAction implements IElementAction
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractElementAction.class);
+
+    @Inject private IWaitActions waitActions;
```

Comment:
use constructor?

Reply:
refactoring this to constructors will spawn approximately 30 actions with 3-5 arg constructors which makes refactoring or adding new service beans to this class really hard

Reply:
Resolved